src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java

```diff
@@ -139,33 +176,43 @@ private boolean isBaseTypeValidator(FieldValidator validator, BaseType fieldType
         return false;
     }
 
-    private List<ValidationResult> validateCollectionItem(FieldTypeDefinition fieldTypeDefinition, JsonNode item,
-                                                          String fieldIdPrefix, String index) {
+    private List<ValidationResult> validateCollectionItem(
+        FieldTypeDefinition fieldTypeDefinition,
+        JsonNode item,
+        String fieldIdPrefix,
+        String index,
+        ValidationContext validationContext
+    ) {
         final String itemFieldId = fieldIdPrefix + index;
 
         final JsonNode itemValue = item.get(CollectionValidator.VALUE);
 
         if (null == itemValue) {
             return Collections.emptyList();
         }
-
         if (shouldTreatAsValueNode(fieldTypeDefinition, itemValue)) {
             if (!BaseType.contains(fieldTypeDefinition.getType())) {
-                return Collections.singletonList(new ValidationResult("Unknown Type:"
-                    + fieldTypeDefinition.getType(), itemFieldId));
+                return Collections.singletonList(
+                    new ValidationResult("Unknown Type:" + fieldTypeDefinition.getType(), itemFieldId)
+                );
             }
-
             final BaseType baseType = BaseType.get(fieldTypeDefinition.getType());
-
             final CaseFieldDefinition caseFieldDefinition = new CaseFieldDefinition();
             caseFieldDefinition.setFieldTypeDefinition(fieldTypeDefinition);
             caseFieldDefinition.setId(index);
-            return validateSimpleField(index, itemValue, caseFieldDefinition, fieldIdPrefix, baseType);
+            return validateSimpleField(
+                                        index,
+                                        itemValue,
+                                        caseFieldDefinition,
+                                        fieldIdPrefix,
+                                        baseType,
+                                        validationContext
```

Comment:
validationContext.getCaseFieldDefinition can be used, no need to pass caseFieldDefinition parameter

Reply:
mmmmm No we are trying to void it , cos these values can change among the multiple iterations of the recursive calls in CaseDataValidator.java. that is why we keep  setting.